src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -706,9 +706,10 @@ private BoundExpression CreateAnonymousFunctionConversion(SyntaxNode syntax, Bou
             // UNDONE: is converted to a delegate that does not match. What to surface then?
 
             var unboundLambda = (UnboundLambda)source;
-
             var boundLambda = unboundLambda.Bind((NamedTypeSymbol)destination, isExpressionTree: destination.IsGenericOrNonGenericExpressionType(out _));
             diagnostics.AddRange(boundLambda.Diagnostics);
+
+            bool hasErrors = CheckValidScopedMethodConversion(syntax, boundLambda.Symbol, destination, invokedAsExtensionMethod: false, diagnostics);
```

Comment:
>hasErrors [](http://example.com/codeflow?start=17&length=9)

I wouldn't mark node with hasErrors flag because of this. The purpose of the flag is mostly to suppress cascading diagnostics for completely broken scenarios. When we cannot accurately figure out the result type, etc. It is not required to reflect if there are any errors associated with the node. In this case, it doesn't look like the new error can cause a cascading diagnostics.